### PR TITLE
HPCC-23896 Missing ESP logging in cloud

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x]  I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [x] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: system/jlib/jlog.cpp [hunk 1]
```diff
@@ -2350,6 +2351,11 @@ void setupContainerizedLogMsgHandler()
     IPropertyTree * logConfig = queryComponentConfig().queryPropTree("logging");
     if (logConfig)
     {
+        if (logConfig->getPropBool(logQueueDisabledAtt, false))
```
Comment (jakesmith) [inline system/jlib/jlog.cpp:2354]: perhaps should also disable if traceLevel = 0?

File: roxie/ccd/ccdmain.cpp [hunk 2]
```diff
@@ -715,6 +715,8 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
         traceLevel = topology->getPropInt("@traceLevel", runOnce ? 0 : 1);
         if (traceLevel > MAXTRACELEVEL)
             traceLevel = MAXTRACELEVEL;
+        if (traceLevel && topology->hasProp("logging/@disabled"))
+            topology->setPropBool("logging/@disabled", false);
```
Comment (jakesmith) [inline roxie/ccd/ccdmain.cpp:719]: If a config file had disabled=true, I'd have thought that would override other logging settings like traveLevel.
But this means if traveLevel>0, it ignores @disabled flag.

Is this logic required?

**Comment by jakesmith** [comment]
@richardkchapman - looks good, a couple of questions.

File: system/jlib/jlog.cpp (hunk 1 above)
Comment (richardkchapman) [reply system/jlib/jlog.cpp:2354]: traceLevel is not something known about at jlib level (but yes, whole area of traceLevel vs loggingDetail is a bit of a mess)

File: roxie/ccd/ccdmain.cpp (hunk 2 above)
Comment (richardkchapman) [reply roxie/ccd/ccdmain.cpp:719]: It's so that I can set traceLevel=1 when running a standalone program (where default logging is disabled)

**Comment by richardkchapman** [comment]
@jakesmith Questions answered

@ghalliday Please review/merge

**Pull request merged**